packages/core/src/store/deal-store.js

```diff
@@ -0,0 +1,174 @@
+ * @typedef {import('./types').DealStoreRecordKey} DealStoreRecordKey
+ * @typedef {import('./types').DealStoreRecordQueryByPiece} DealStoreRecordQueryByPiece
+ */
+
+/**
+ * @param {DealRecord} record 
+ * @returns {DealStoreRecord} 
+ */
+const encodeRecord = (record) => {
+  return {
+    ...record,
+    piece: record.piece.toString()
+  }
+}
+
+/**
+ * @param {DealRecordKey} recordKey 
+ * @returns {DealStoreRecordKey} 
+ */
+const encodeKey = (recordKey) => {
+  return {
+    ...recordKey,
+    piece: recordKey.piece.toString()
+  }
+}
+
+/**
+ * @param {DealRecordQueryByPiece} recordKey 
+ * @returns {DealStoreRecordQueryByPiece} 
+ */
+const encodeQueryByPiece = (recordKey) => {
+  return {
+    ...recordKey,
+    piece: recordKey.piece.toString()
+  }
+}
+
+/**
+ * @param {DealStoreRecord} encodedRecord 
+ * @returns {DealRecord}
+ */
+const decodeRecord = (encodedRecord) => {
+  return {
+    ...encodedRecord,
+    piece: parseLink(encodedRecord.piece)
+  }
+}
+
+/**
+ * @param {import('./types.js').TableConnect | import('@aws-sdk/client-dynamodb').DynamoDBClient} conf
+ * @param {object} context
+ * @param {string} context.tableName
+ * @returns {import('@web3-storage/filecoin-api/deal-tracker/api').DealStore}
+ */
+export function createClient (conf, context) {
+  const tableclient = connectTable(conf)
+
+  return {
+    put: async (record) => {
+      const putCmd = new PutItemCommand({
+        TableName: context.tableName,
+        Item: marshall(encodeRecord(record), {
+          removeUndefinedValues: true
+        }),
+      })
+
+      try {
+        await tableclient.send(putCmd)
+      } catch (/** @type {any} */ error) {
+        return {
+          error: new StoreOperationFailed(error.message)
+        }
+      }
+
+      return {
+        ok: {}
+      }
+    },
+    get: async (key) => {
+      const getCmd = new GetItemCommand({
+        TableName: context.tableName,
+        Key: marshall(encodeKey(key))
+      })
+      let res
+      try {
+        res = await tableclient.send(getCmd)
+      } catch (/** @type {any} */ error) {
+        return {
+          error: new StoreOperationFailed(error.message)
+        }
+      }
+
+      // not found error
+      if (!res.Item) {
+        return {
+          error: new RecordNotFound('item not found in store')
+        }
+      }
+
+      return {
+        ok: decodeRecord(
+          /** @type {DealStoreRecord} */ (unmarshall(res.Item))
+        )
+      }
+    },
+    has: async (key) => {
+      const getCmd = new GetItemCommand({
+        TableName: context.tableName,
+        Key: marshall(encodeKey(key))
+      })
+      let res
+      try {
+        res = await tableclient.send(getCmd)
+      } catch (/** @type {any} */ error) {
+        return {
+          error: new StoreOperationFailed(error.message)
+        }
+      }
+
+      // not found error
+      if (!res.Item) {
+        return {
+          error: new RecordNotFound('item not found in store')
```

Comment:
Hmm, personally I'd expect `has` to return `true`/`false`. It's not an error state if the record is not found when you're asking it if it has the record or not...